exec/vector/src/main/java/org/apache/drill/exec/vector/complex/FieldIdUtil.java

```diff
@@ -121,4 +125,63 @@ public static TypedFieldId getFieldIdIfMatches(ValueVector vector, TypedFieldId.
       }
     }
   }
+
+  public static TypedFieldId getFieldId(ValueVector vector, int id, SchemaPath expectedPath, boolean hyper) {
```

Comment:
Is it possible to share this code (or much of it with getFieldIdIfMatches() above). It seems like they are related (although I didn't go line by line)